src/gui/folderman.cpp

```diff
@@ -837,6 +845,22 @@ bool FolderMan::pushNotificationsFilesReady(Account *account)
     return pushFilesAvailable && pushNotifications && pushNotifications->isReady();
 }
 
+bool FolderMan::switchToVfsIsNeeded(const FolderDefinition &folderDefinition) const
+{
+    auto result = false;
+    if (ENFORCE_VIRTUAL_FILES_SYNC_FOLDER) {
+        const auto bestVfsPlugin = bestAvailableVfsMode();
+
+        if (folderDefinition.virtualFilesMode != bestVfsPlugin &&
+                folderDefinition.virtualFilesMode == Vfs::Off &&
+                OCC::Theme::instance()->showVirtualFilesOption()) {
+            result = true;
```

Comment:
Nit: why not simply return true and then get rid of the `result` variable? Below you can simply return `false`.

Reply:
my habit of doing RVO always (so many return but single variable always
and OK that does not make sense for `bool` data type